roster/src/components/ShiftAssignmentDialog.vue

```diff
@@ -0,0 +1,458 @@
+<template>
+	<Dialog :options="{ title: dialog.title, size: '4xl' }">
+		<template #body-content>
+			<div class="grid grid-cols-2 gap-6">
+				<FormControl
+					type="autocomplete"
+					label="Employee"
+					v-model="form.employee"
+					:disabled="!!props.shiftAssignmentName"
+					:options="employees"
+				/>
+				<FormControl type="text" label="Company" v-model="form.company" :disabled="true" />
+				<FormControl
+					type="text"
+					label="Employee Name"
+					v-model="form.employee_name"
+					:disabled="true"
+				/>
+				<FormControl
+					type="date"
+					label="Start Date"
+					v-model="form.start_date"
+					:disabled="!!props.shiftAssignmentName"
+				/>
+				<FormControl
+					type="autocomplete"
+					label="Shift Type"
+					v-model="form.shift_type"
+					:disabled="!!props.shiftAssignmentName"
+					:options="shiftTypes.data"
+				/>
+				<FormControl type="date" label="End Date" v-model="form.end_date" />
+				<FormControl
+					type="select"
+					:options="['Active', 'Inactive']"
+					label="Status"
+					v-model="form.status"
+				/>
+				<FormControl
+					type="text"
+					label="Department"
+					v-model="form.department"
+					:disabled="true"
+				/>
+				<div v-if="!props.shiftAssignmentName">
+					<FormControl
+						type="checkbox"
+						label="Select Working Days"
+						v-model="selectDays"
+						:disabled="disableSelectDays"
+					/>
+				</div>
+				<div v-if="!props.shiftAssignmentName" />
+				<div v-if="!props.shiftAssignmentName && selectDays" class="space-y-1.5">
+					<div class="text-xs text-gray-600">Days</div>
+					<div
+						class="border rounded grid grid-flow-col h-7 justify-stretch overflow-clip"
+					>
+						<div
+							v-for="(isSelected, day) of workingDays"
+							class="cursor-pointer flex flex-col"
+							:class="{
+								'border-r': day !== 'Sunday',
+								'bg-gray-100 text-gray-500': !isSelected,
+							}"
+							@click="workingDays[day] = !workingDays[day]"
+						>
+							<div class="text-center text-sm my-auto">
+								{{ day.substring(0, 3) }}
+							</div>
+						</div>
+					</div>
+				</div>
+				<FormControl
+					v-if="!props.shiftAssignmentName && selectDays"
+					type="select"
+					:options="['Every Week', 'Every 2 Weeks', 'Every 3 Weeks', 'Every 4 Weeks']"
+					label="Frequency"
+					v-model="frequency"
+				/>
+			</div>
+		</template>
+		<template #actions>
+			<div class="flex space-x-2 justify-end">
+				<Dropdown
+					v-if="props.shiftAssignmentName"
+					:options="actions"
+					:button="{ size: 'md', icon: 'more-vertical' }"
```

Comment:
These options are confusing as to what effect they have on the backend.
Delete current shift - only deletes that day's shift. Instead of this can we call it something else or maybe have a description? 
Delete consecutive shifts - this deletes the shift assignment
The last option deletes the entire schedule

All these should have more description also add a confirmation dialog on the action 
"This will delete Shift Assignment {0} from date A to date B"
and so on. 
Add links to open the shift assignment doc/shift schedule doc in a new tab from the message

![image](https://github.com/frappe/hrms/assets/24353136/34c5ab37-e307-4667-a95c-a68481b25238)

Reply:
We should also validate more here. Eg: if they are trying to delete an old shift assignment whose check-ins and attendance have already been created then we shouldn't allow deletion.